plugins/follow_me/follow_me.h

```diff
@@ -0,0 +1,178 @@
+#include <cmath>
+#include <functional>
+
+namespace dronecore {
+
+class FollowMeImpl;
+
+/**
+ * @brief This class enables vehicle tracking of a specified target (typically a ground station carried by a user).
+ *
+ * The API is used to supply both the position(s) for the target and the relative follow position of the vehicle.
+ * Applications must get target position information from the underlying platform (or some other source).
+ *
+ * See https://docs.px4.io/en/flight_modes/follow_me.html for more information.
+ *
+ */
+class FollowMe
+{
+public:
+    /**
+     * @brief Constructor (internal use only).
+     * @param impl Private internal implementation.
+     */
+    explicit FollowMe(FollowMeImpl *impl);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~FollowMe();
+
+    /**
+     * @brief Results of FollowMe operations.
+     */
+    enum class Result {
+        SUCCESS = 0, /**< @brief Request succeeded. */
+        NO_DEVICE, /**< @brief No device connected. */
+        CONNECTION_ERROR, /**< @brief %Connection error. */
+        BUSY, /**< @brief Vehicle busy. */
+        COMMAND_DENIED, /**< @brief Command denied. */
+        TIMEOUT, /**< @brief Request timeout. */
+        NOT_ACTIVE, /**< @brief FollowMe is not activated. */
+        UNKNOWN /**< @brief Unknown error. */
+    };
+
+    /**
+     * @brief FollowMe Configuration.
+     * Reference: https://docs.px4.io/en/advanced_config/parameter_reference.html#follow-target
+     */
+    struct Config {
+        /**
+         * @brief Relative position of following vehicle.
+         */
+        enum class FollowDirection {
+            FRONT_RIGHT, /**< @brief Follow from front right. */
+            BEHIND, /**< @brief Follow from behind. */
+            FRONT, /**< @brief Follow from front. */
+            FRONT_LEFT, /**< @brief Follow from front left. */
+            NONE
+        };
+
+        constexpr static const float DEFAULT_HEIGHT_M =
+            8.0; /**< @brief Default height above home, in meters. */
+        constexpr static const float DEFAULT_FOLLOW_DIST_M =
+            8.0; /**< @brief Default follow distance, in meters. */
+        constexpr static const FollowDirection DEFAULT_FOLLOW_DIR =
+            FollowDirection::BEHIND; /**< @brief Default follow direction. */
+        constexpr static const float DEFAULT_RESPONSIVENSS =
+            0.5f; /**< @brief Default responsiveness to target. */
+
+        constexpr static const float MIN_HEIGHT_M = 8.0f; /**< @brief Min follow height, in meters. */
+        constexpr static const float MIN_FOLLOW_DIST_M =
+            1.0f; /**< @brief Min follow distance from, in meters. */
+        constexpr static const float MIN_RESPONSIVENESS = 0.f; /**< @brief Min responsiveness. */
+        constexpr static const float MAX_RESPONSIVENESS = 1.0f; /**< @brief Max responsiveness. */
+
+        float min_height_m = DEFAULT_HEIGHT_M; /**< @brief Min follow height above home, in meters. */
```

Comment:
What about just setting the number here instead of having the #define? Do you think it is needed?

Reply:
Yeah, it is not used anywhere. Initially I had used these constants for future use, if we need default values, etc. It seems we don't require them. I shall remove them and use literals.